robot/impl/robot_reconfigure_test.go

```diff
@@ -3754,16 +3754,64 @@ func TestResourceConstructCtxCancel(t *testing.T) {
 	})
 }
 
+func TestResourceCloseNoHang(t *testing.T) {
+	logger := logging.NewTestLogger(t)
+
+	mockAPI := resource.APINamespaceRDK.WithComponentType("mock")
+	modelName1 := utils.RandomAlphaString(5)
+	model1 := resource.DefaultModelFamily.WithModel(modelName1)
+
+	resource.RegisterComponent(mockAPI, model1, resource.Registration[resource.Resource, resource.NoNativeConfig]{
+		Constructor: func(
+			ctx context.Context,
+			deps resource.Dependencies,
+			conf resource.Config,
+			logger logging.Logger,
+		) (resource.Resource, error) {
+			return &mockFake{Named: conf.ResourceName().AsNamed(), shouldHangOnClose: true}, nil
+		},
+	})
+	defer func() {
+		resource.Deregister(mockAPI, model1)
+	}()
+
+	cfg := &config.Config{
+		Components: []resource.Config{
+			{
+				Name:  "mock",
+				Model: model1,
+				API:   mockAPI,
+			},
+		},
+	}
+	r, err := New(context.Background(), cfg, logger)
+	test.That(t, err, test.ShouldBeNil)
+
+	// Lower resourceCloseTimeout to one second to speed up test.
+	originalResourceCloseTimeout := resourceCloseTimeout
```

Comment:
maybe the test should be construct closer to how `TestResourceConstructTimeout` is made, so we only check that the timeout/deadline is what we expect

Reply:
Hmm I haven't thought _super_ hard about it, but my intuition is that asserting anything in the `Close` method of `mockFake` may be difficult as we don't have access to the testing object to make `test.That` calls about the state of the context deadline. How do you feel about @maximpertsov's suggestion to just shorten to a minute instead of an hour?

Reply:
we could return a precreated mockFake in the constructor, and then have Close() set a struct var for the deadline that was passed in

Reply:
Ah gotcha! Yeah much simpler thanks for explaining.